EIPS/eip-7870.md

```diff
@@ -0,0 +1,146 @@
+author: Parithosh Jayanthi (@parithosh), Kevaundray Wedderburn (@kevaundray), Josh Rudolf (@jrudolf), Dankrad Feist (@dankrad), Justin Traglia (@jtraglia), Ignacio Hagopian (@jsign), George Kadianakis (@asn-d6), Fredrik Svantes (@fredriksvantes)
+discussions-to: https://ethereum-magicians.org/t/hardware-and-bandwidth-recommendations-for-full-nodes-and-validators/22675
+status: Draft
+type: Informational
+created: 2025-01-26
+---
+
+## Abstract
+
+This proposal specifies hardware and bandwidth recommendations for different types of Ethereum nodes:
+
+- **Full nodes**: Nodes that follow the tip of the chain without necessarily proposing blocks.
+- **Validators**: Split into:
+  - **Attesters**: Validators that validate and attest to blocks created by proposers.
+  - **Local block builders** (Proposers): Validators that create blocks locally and broadcast them to the network.
+
+The resource-intensive aspect for local block builders lies in creating the block and quickly broadcasting the data required for attesters to validate it in time.
+
+We note that it may be possible to run a client with less than the recommended specifications, however benchmarks and decision-making will be made with respect to these recommendations.
+
+## Motivation
+
+Clear system specifications are crucial for:
+
+- Ensuring meaningful benchmark comparisons across different client implementations.
+- Enabling informed decision-making about protocol upgrades and their resource usage implications.
+- Providing clear guidance for node operators to ensure alignment with future network requirements.
+
+Without a shared understanding of target hardware specifications:
+
+- Benchmark results lose significance due to inconsistent testing environments.
+- Decision-making becomes challenging for implementation choices, as performance characteristics are heavily hardware-dependent.
+- Network participants lack clear guidance for hardware investments.
+
+## Specification
+
+### Roles and Their Recommended Specifications
+
+Node operators typically run both an **Execution Layer (EL)** client and a **Consensus Layer (CL)** client on the same machine. The specifications below assume the combined resource usage of both.
+
+| Node Type               | Storage       | RAM   | CPU Cores / Threads | **PassMark CPU Rating**                            | Download / Upload speed  |
+|-------------------------|---------------|-------|---------------------|----------------------------------------------------|--------------------------|
+| **Full Node**           | 4 TB NVMe     | 32 GB | 4c / 8t             | ~1000 ST / 3000 MT                                 | 50 Mbps / 15 Mbps        |
+| **Attester**            | 4 TB NVMe     | 64 GB | 8c / 16t            | ~3500 ST / 25000 MT                                | 50 Mbps / 25 Mbps        |
+| **Local Block Builder** | 4 TB NVMe     | 64 GB | 8c / 16t            | ~3500 ST / 25000 MT                                | 100 Mbps / 50 Mbps       |
```

Comment:
The table needs to be re-formatted again 😄

Reply:
Also, I think we should change "RAM" in the title to "Memory".

Reply:
Also "speed" in "Download / Upload speed" is lowercase, when it probably shouldn't be. I'd be in favor of renaming this title to "Bandwidth Download / Upload".